### ISSUE TITLE
alpm: can not install updates

### ISSUE THREAD

**Issue by eworm-de** [hughsie/PackageKit#32]
After the update to libalpm 9.0.0 / pacman 4.2.0 installing updates with packagekit fails.

gnome-packagekit lists the available updates, pressing "Install updates" makes packagekit work for some time, then it returns with the same available updates. No packages have been upated.

Even pkcon behaves strange:

    root@08-00-27-f1-27-24 ~ # pkcon get-details nftables
    Resolving                     [=========================]         
    Getting details               [=========================]         
    Finished                      [=========================]         
    Package description
      package:     nftables-1:0.4-2.x86_64
      summary:     (null)
      license:     GPL2
      group:       other
      description: Netfilter tables userspace tools
      size:        328704 bytes
      url:         http://netfilter.org/projects/nftables/
    root@08-00-27-f1-27-24 ~ # pkcon update nftables
    Resolving                     [                         ] (0%)  Command failed: This tool could not find the package: nftables: could not find or read package

**Comment by fbourigault** [comment]
Confirmed here. Systemd offline updates are working well.

**Comment by fbourigault** [comment]
I looked in the code how was handled package upgrade and I saw that packages install and packages update are handled by two different functions.

As in alpm world, installing or updating a package is the same (sync) I give a try to just call `pk_backend_install_packages_thread` instead of `pk_backend_update_packages_thread` in `pk_backend_update_packages`.

By doing this, I just disabled the download only feature which can easily be added to the install function (and it's a valid behavior of pacman for first time package install). I also disabled something more complex to describe : when a package is installed during an upgrade because it replaces a newly deprecated one, this new package was marked as explicitly installed if the removed one was explicitly installed. I don't know if this behavior is wanted but it can be temporary removed and re-introduced after  some refactoring.

So now when running `./pkcon update` I get the following output :
```
Getting updates               [=========================]         
Finished                      [=========================]         
Downloading list of packages  [=========================]         
Testing changes               [=========================]         
Finished                      [                         ] (0%)  
The following packages have to be updated:
 libjpeg-turbo-1.4.0-1.x86_64	JPEG image codec with accelerated baseline compression and decompression
 openssl-1.0.1.k-1.x86_64	The Open Source toolkit for Secure Sockets Layer and Transport Layer Security
 ttf-liberation-2.00.1-5.any	Red Hats Liberation fonts.
 unzip-6.0-8.x86_64	For extracting and viewing files in .zip archives
 xz-5.0.8-1.x86_64	Library and command line tools for XZ and LZMA compressed files
Proceed with changes? [N/y] N

The transaction did not proceed.
                              [=========================]         
Fatal error: user declined simulation
``` 

I was happy and decided to decline the update. After I ran the same command and I get the following output :
```
Getting updates               [=========================]         
Finished                      [=========================]         
No packages require updating to newer versions.
```

I can confirm there is nothing to install anymore, packages were upgraded. So when I was running the simulation, packages where really installed.

This lead me to a question : does simulation something required within PackageKit (cc @hughsie) ?

**Comment by fbourigault** [comment]
Fixed since #38 is merged into master.
